src/test/java/hudson/plugins/parameterizedtrigger/test/BuildTriggerTest.java

```diff
@@ -70,7 +68,18 @@ public void testParentProjectTrigger() throws Exception {
         upstream.getPublishersList().add(triggerBuilder);
 
         r.buildAndAssertSuccess(upstream);
-        Awaitility.await().pollInterval(1, SECONDS).atMost(10, SECONDS).until(() -> downstream.getLastBuild() != null);
+        r.waitUntilNoActivity();
+
+        if (downstream.getLastBuild() == null) {
+            /*
+             * There could be a period when upstream already finished, but async rebuilding graph {@link Jenkins#rebuildDependencyGraphAsync()}
+             * could still be in progress or even not start.
+             * Let's rebuild graph manually in that case to avoid flakiness.
+             */
+            r.jenkins.rebuildDependencyGraph();
```

Comment:
would it not be better to do this before the build was even started? and then you avoid the whole need for another build.  (which is what the tests in Jenkins core actually [appear to do](https://github.com/jenkinsci/jenkins/search?l=Java&q=rebuildDependencyGraph)?

are there other places in this plugins tests that also show the same issues that should be fixed also, or do they already rebuild the dependency graph correctly?

Reply:
>would it not be better to do this before the build was even started? and then you avoid the whole need for another build. (which is what the tests in Jenkins core actually [appear to do](https://github.com/jenkinsci/jenkins/search?l=Java&q=rebuildDependencyGraph)?

Yes it's a good option, this is actually what most of the test already do. (changes in [3ea7129](https://github.com/jenkinsci/parameterized-trigger-plugin/pull/266/commits/3ea7129a9ec00173db196122552f9f4402e0896d))

>are there other places in this plugins tests that also show the same issues that should be fixed also, or do they already rebuild the dependency graph correctly?

Most of the test call `r.jenkins.rebuildDependencyGraph();` but not all of them, I will investigate if there could be something related.